README.md

```diff
@@ -1,3 +1,134 @@
+* and some common implementations to get everyone stated
+
+(unpresented) ROSCon 2018 Lightning Talk [slides](docs/fuse_lightning_talk.pdf)
+
+
+Data flows through the system approximately like this:
+* A sensor model receives raw sensor data. The sensor model generates a constraint and sends it to the optimizer.
+* The optimizer receives the new sensor constraint. A request is sent to each configured motion model to generate
+  a constraint between the previous state and the new state involved in the sensor constraint.
+* The motion model receives the request and generates the required constraints to connect the new state to the
+  previously generated motion model chain. The motion model constraints are sent to the optimizer.
+* The optimizer adds the new sensor model and motion model constraints and variables to the graph and
+  computes the optimal values for each state variable.
+* The optimal state values are sent to each configured publisher (as well as the sensor models and motion models).
+* The publishers receive the optimized state values and publish any derived quantities on ROS topics.
+* Repeat
+
+It is important to note that much of this flow happens asynchronously in practice. Sensors are expected to operate
+independently from each other, so each sensor will be sending constraints to the optimizer at its own frequency. The
+optimizer will (probably) cache the constraints and process them in small batches on some schedule. The publishers may
+require considerable processing time, introducing a delay between the completion of the optimization cycle and the
+publishing of data to the ROS topic.
+
+![fuse sequence diagram](doc/fuse_sequence_diagram.png)
+
+## Example
+Let's consider a simple robotics example to illustrate this. Suppose we have a typical indoor differential-drive robot.
+This robot will have wheel encoders and a horizontal laser.
+
+The first thing we must do is define our state variables. At a minimum, we will want the robot pose at each timestamp.
+Let's assume we model the pose using two different state variables, a 2D position and orientation `(x, y, yaw)`.
+Each 2D pose _at a specific time_ will get a unique variable name. For ease of notation, let's call the pose
+variables `X1`, `X2`, `X3`, etc. (In reality, each variable will get a UUID, but those are much harder to write down.)
+We will need to derive a 2D pose class from the `fuse_core::Variable` base class. (Actually, `fuse` ships with a 2D
+position and 2D orientation class, but you get the idea. You can create your own variable types when you need them.)
+
+Next we need to decide how to model our sensors. We will probably model the wheel encoders as providing an incremental
+pose measurement. Given a starting pose, `X1` and a wheel encoder delta `z`, you can predict the pose `X2`. So we end
+up with an error term that looks something like `error = f(X1, z)^-1 * X2`. So we need to derive a
```

Comment:
Is this a square root?

Reply:
No, it's more of a rigid transform inverse.
`X2' = f(X1, z)`
`error = X2'^-1 * X2`
`error` is the delta between the predicted `X2'` and the actual `X2`.